Lib/reprlib.py

```diff
@@ -60,15 +71,33 @@ def repr(self, x):
         return self.repr1(x, self.maxlevel)
 
     def repr1(self, x, level):
-        typename = type(x).__name__
-        if ' ' in typename:
+        _type = type(x)
+        typename = _type.__name__
+
+        if " " in typename:
             parts = typename.split()
-            typename = '_'.join(parts)
-        if hasattr(self, 'repr_' + typename):
-            return getattr(self, 'repr_' + typename)(x, level)
-        else:
+            typename = "_".join(parts)
+
+        method_name = "repr_" + typename
+
+        if not hasattr(self, method_name):
             return self.repr_instance(x, level)
 
+        # we have a predefined method for this type,
+        # but it has been set in a subclass
+        if method_name not in self._lookup:
+            return getattr(self, method_name)(x, level)
```

Comment:
You use `hasattr` with the same args right above. Please, refactor these two calls into one.